src/natcap/invest/urban_flood_risk_mitigation.py

```diff
@@ -322,107 +313,245 @@ def execute(args):
-
-    target_watershed_result_vector_path = os.path.join(
-        args['workspace_dir'], 'flood_risk_service%s.shp' % file_suffix)
+            (runoff_retention_raster_path, 1),
+            reprojected_aoi_path),
+        dependent_task_list=[runoff_retention_task],
+        task_name='zonal_statistics over runoff_retention raster')
 
-    task_graph.add_task(
-        func=_add_zonal_stats,
+    runoff_retention_volume_stats_task = task_graph.add_task(
+        func=pygeoprocessing.zonal_statistics,
         args=(
-            runoff_retention_pickle_path,
-            runoff_retention_ret_vol_pickle_path,
-            flood_vol_pickle_path,
-            intermediate_target_watershed_result_vector_path,
-            target_watershed_result_vector_path),
-        target_path_list=[target_watershed_result_vector_path],
-        dependent_task_list=[
-            flood_vol_pickle_task, runoff_retention_ret_vol_pickle_task,
-            runoff_retention_pickle_task, intermediate_affected_vector_task],
-        task_name='add zonal stats')
+            (runoff_retention_vol_raster_path, 1),
+            reprojected_aoi_path),
+        dependent_task_list=[runoff_retention_ret_vol_task],
+        task_name='zonal_statistics over runoff_retention_volume raster')
+
+    damage_per_aoi_stats = None
+    flood_volume_stats = None
+    summary_tasks = [
+            runoff_retention_stats_task,
+            runoff_retention_volume_stats_task]
+    if 'built_infrastructure_vector_path' in args and (
+            args['built_infrastructure_vector_path'] not in ('', None)):
+        # Reproject the built infrastructure vector to the target SRS.
+        reprojected_structures_path = os.path.join(
+                intermediate_dir, 'structures_reprojected.gpkg')
+        reproject_built_infrastructure_task = task_graph.add_task(
+            func=pygeoprocessing.reproject_vector,
+            args=(args['built_infrastructure_vector_path'],
+                  target_sr_wkt,
+                  reprojected_structures_path),
+            kwargs={'driver_name': 'GPKG'},
+            target_path_list=[reprojected_structures_path],
+            task_name='reproject built infrastructure to target SRS')
+
+        # determine the total damage to all infrastructure in the watershed/AOI
+        damage_to_infrastructure_in_aoi_task = task_graph.add_task(
+            func=_calculate_damage_to_infrastructure_in_aoi,
+            args=(reprojected_aoi_path,
+                  reprojected_structures_path,
+                  args['infrastructure_damage_loss_table_path']),
+            dependent_task_list=[
+                reprojected_aoi_task,
+                reproject_built_infrastructure_task],
+            task_name='calculate damage to infrastructure in aoi')
+        summary_tasks.append(damage_to_infrastructure_in_aoi_task)
+
+        # Determine flood_volume over the watershed
+        flood_volume_in_aoi_task = task_graph.add_task(

```

Comment:
This is an interesting one!  I added the tasks in f9533e26 for good measure, but also they shouldn't be strictly necessary.  Because all 4 of the tasks tracked in `summary_tasks` have their `.get()` method called, which will block until the task completes.  I added them anyways as a 'just in case' sort of condition, and also because I suppose that it might help with determining whether the task should be recomputed.

Reply:
Oh cool, I did not know about that feature of `get()`.